src/test/java/bookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,229 @@
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        String responseContent = result.getResponse().getContentAsString();
+        JsonNode rootNode = objectMapper.readTree(responseContent);
+        JsonNode dataNode = rootNode.get("data");
+
+        assertNotNull(dataNode);
+
+        CategoryDto actual = objectMapper.treeToValue(dataNode, CategoryDto.class);
+
+        assertNotNull(actual);
+        Assertions.assertEquals(expected.getName(), actual.getName());
+        Assertions.assertEquals(expected.getDescription(), actual.getDescription());
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get category by ID")
+    void getCategoryById_ReturnsCategoryById_ExpectedSuccess() throws Exception {
+        Long categoryId = 1L;
+        MvcResult result = mockMvc.perform(
+                        get("/categories/{id}", categoryId)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        CategoryDto category = objectMapper
+                .readValue(result.getResponse().getContentAsString(), CategoryDto.class);
+        assertNotNull(category);
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Update a category")
+    void updateCategory_UpdatesCategoryDetails_ExpectedSuccess() throws Exception {
+        CategoryDto categoryDto = new CategoryDto();
+        categoryDto.setId(1L);
+        categoryDto.setName("Updated Category Name");
+        categoryDto.setDescription("Updated Category Description");
+
+        String jsonRequest = objectMapper.writeValueAsString(categoryDto);
+
+        MvcResult result = mockMvc.perform(
+                        put("/categories/{id}", 1L)
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+
+        mockMvc.perform(
+                        put("/categories/{id}", 1L)
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.data.name").value("Updated Category Name"))
+                .andExpect(jsonPath("$.data.description").value("Updated Category Description"));
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Delete a category")
+    void deleteCategory_DeletesCategoryById_ExpectedSuccess() throws Exception {
```

Comment:
Omg)
The same test for delete method in BookControllerTest looks differently. Please, see how you did it in BookControllerTest